### PR TITLE
Fix OSX requirements

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xhochy** [comment]
@conda-forge/watchdog This is ready for review/merge.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,17 +28,21 @@ requirements:
...
+  requires:
+    - pip
   imports:
     - watchdog
     - watchdog.observers
+    - watchdog.observers.fsevents  # [osx]
     - watchdog.tricks
     - watchdog.utils
   commands:
+    - python -m pip check
```
Comment (ccordoba12) [inline recipe/meta.yaml:47]: What does this test do?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,17 +28,21 @@ requirements:
...
     - watchmedo --help
-    - test -f $SP_DIR/_watchdog_fsevents.so  # [osx and py2k]
-    - test -f $SP_DIR/_watchdog_fsevents.cpython-37m-darwin.so  # [osx and py==37]
```
Comment (ccordoba12) [inline recipe/meta.yaml:41]: Why did you remove these tests?

Comment (xhochy) [reply recipe/meta.yaml:41]: I did replace them with an import test (forgot to push) and added a C compiler as a requirement as this is used during the build of them.

**Comment by xhochy** [comment]
@ccordoba12 Can you have another look, please? Comments are adressed and CI passed again.

File: recipe/meta.yaml (hunk 2 above)
Comment (goanpeca) [reply recipe/meta.yaml:41]: I do not think pip check replaces these checks

File: recipe/meta.yaml [hunk 3]
```diff
@@ -28,17 +30,22 @@ requirements:
...
 test:
+  requires:
+    - pip
   imports:
     - watchdog
     - watchdog.observers
+    - watchdog.observers.fsevents  # [osx]
     - watchdog.tricks
     - watchdog.utils
+    - _watchdog_fsevents  # [osx]
```
Comment (xhochy) [inline recipe/meta.yaml:45]: @goanpeca This is the replacement for `test -f $SP_DIR/_watchdog_fsevents.`

File: recipe/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/meta.yaml:47]: Checks for the consistency of the Python requirements as declared by the package. This will fail if a dependency was forgotten, in the case of `watchdog`, it would have errored on the missing `pyobjc-framework-cocoa` on OSX.

File: recipe/meta.yaml (hunk 2 above)
Comment (xhochy) [reply recipe/meta.yaml:41]: @goanpeca No, they are replaced by an import check as above. These are Python modules where the so-name depends on the Python version, the Python module name is though version independent.

**Review by goanpeca — APPROVED** [review]
I see! Thanks for the explanation
